cookietemple/create/template_creator.py

```diff
@@ -293,22 +283,43 @@ def create_common_files(self) -> None:
         # change to recent cwd so lint etc can run properly
         os.chdir(str(cwd_project))
 
-    def readthedocs_slug_already_exists(self, project_name: str) -> bool:
+    def check_pypi_readthedocs(self, host, dot_cookietemple) -> None:
+        """
+        Main function that calls the checks for PyPi and readthedocs
+        """
+        while TemplateCreator.call_host(host, self.creator_ctx.project_name) and not dot_cookietemple:
+            print(f'[bold red]A project named {self.creator_ctx.project_name} already exists at {host}!')
+            if cookietemple_questionary_or_dot_cookietemple(function='confirm',
+                                                            question='Do you want to choose another name for your project?\n'
+                                                                     f'Otherwise you will not be able to host your project at {host}!', default='Yes'):
+                self.creator_ctx.project_name = cookietemple_questionary_or_dot_cookietemple(function='text',
+                                                                                             question='Project name',
+                                                                                             default='Exploding Springfield')
+            # break if the project should be named anyways
+            else:
+                break
+
+    @staticmethod
+    def call_host(host: str, project_name: str) -> bool:
```

Comment:
the idea of refactoring this is good, but I had no idea what that name was about when reading it

don't have an idea for a better one atm, but maybe you?

Reply:
Yeah I thought about it as well and was not that's happy, will think again